### PR TITLE
feat: Chatui agent dropdown fields

### PR DESCRIPTION
**Pull request by arduenify**
**Please describe the purpose of this pull request.**
This pull request adds new features to the MemGPT project. Specifically, it implements dropdowns for humans, models, and personas in the React-based chat UI, and introduces corresponding API calls and store setups. The styling of the dropdown menu has been updated to align with the global style of the application.

**How to test**
1. Run `memgpt server` and `npx nx serve`
2. Create a new agent with the default values (don't change the dropdowns)

**Have you tested this PR?**
Here are some screenshots from my tests:

image
image

**Related issues or PRs**
Resolves one of two parts outlined in issue #897. This PR implements the front end and the REST API changes.

**Is your PR over 500 lines of code?**
It is not.

**Additional context**
Note that until the changes are made by @cpacker to the Sync Server, only the default values will work when adding agents, otherwise a 500 is returned. This is also the case for the current production build when typing invalid values.

### CODE REVIEW

File: chatui/libs/components/dropdown-menu/src/dropdown-menu.tsx [hunk 1]
```diff
@@ -6,7 +6,21 @@ import { cn } from '@memgpt/utils';
 
 const DropdownMenu = DropdownMenuPrimitive.Root;
 
-const DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;
```
Comment (goetzrobin) [inline chatui/libs/components/dropdown-menu/src/dropdown-menu.tsx:9]: Let's not add styles here and use the asChild property instead. You can see how it's used to apply the drop down to a button in the RadixUI docs here: https://www.radix-ui.com/primitives/docs/components/dropdown-menu

File: chatui/src/app/modules/home/create-agent-dialog.tsx [hunk 2]
```diff
@@ -53,19 +108,52 @@ const CreateAgentDialog = (props: { open: boolean; onOpenChange: (open: boolean)
 							<Label htmlFor="human" className="text-right">
 								Human
 							</Label>
-							<Input id="human" name="human" defaultValue="cs_phd" className="col-span-3" />
+							<DropdownMenu>
```
Comment (goetzrobin) [inline chatui/src/app/modules/home/create-agent-dialog.tsx:111]: I actually think the select primitive is what we want here: https://ui.shadcn.com/docs/components/select

What do you think?

File: chatui/src/app/modules/home/create-agent-dialog.tsx [hunk 3]
```diff
@@ -7,15 +8,69 @@ import {
...
+    const { data: humansData } = useHumansQuery(auth.uuid);
+    const { data: personasData } = usePersonasQuery(auth.uuid);
+
+    // Zustand stores to manage the state
+    const { setModels } = useModelStore();
+    const { setHumans } = useHumanStore();
+    const { setPersonas } = usePersonaStore();
+
+    // Effect to update the dropdown values when the data is fetched
+    useEffect(() => {
```
Comment (goetzrobin) [inline chatui/src/app/modules/home/create-agent-dialog.tsx:53]: Do you think we need to store this information globally in Zustand? I think just passing the query results to our selects is all we would need. 

TanStack Query already has the results available globally for us!

File: chatui/src/app/modules/home/create-agent-dialog.tsx [hunk 4]
```diff
@@ -7,15 +8,69 @@ import {
...
+			setHumans(humansData.humans);
+			setDropdownValues(prev => ({ ...prev, human: humansData.humans.map(h => h.name) }));
+		}
+		if (Array.isArray(personasData?.personas)) {
+			setPersonas(personasData.personas);
+			setDropdownValues(prev => ({ ...prev, persona: personasData.personas.map(p => p.name) }));
+		}
+	}, [modelsData, humansData, personasData, setModels, setHumans, setPersonas]);
+
+	const [selectedValues, setSelectedValues] = useState({ name: '', human: '', persona: '', model: '' });
```
Comment (goetzrobin) [inline chatui/src/app/modules/home/create-agent-dialog.tsx:68]: We might want to use react-hook-form for managing the form as it is more complicated than just a bunch of inputs now. There's an example in the Memory Update section on what I'm referring to.

File: chatui/libs/components/dropdown-menu/src/dropdown-menu.tsx (hunk 1 above)
Comment (goetzrobin) [reply chatui/libs/components/dropdown-menu/src/dropdown-menu.tsx:9]: Saw you're already using asChild. Sorry about that! What was the reason for adding those styles?

**Review by goetzrobin — COMMENTED** [review]
I left a couple comments. First PR is always hard because of getting used to how we both are doing things! API code looks great and React code is also pretty close already! Great job!

File: chatui/src/app/modules/home/create-agent-dialog.tsx (hunk 3 above)
Comment (goetzrobin) [reply chatui/src/app/modules/home/create-agent-dialog.tsx:53]: Yes please! Agent is a little different. We keep track of the currently selected agent in a global zustand slice of state.

**Review by cpacker — APPROVED** [review]
Thank you @arduenify awesome work!

**Comment by cpacker** [comment]
> @cpacker looks good from my side! I don't have permission to approve PRs but LGTM from my side! 🚀

Thanks for the review @goetzrobin ! I'll try and figure out how to allow you to approve PRs in the future (maybe via some usage of github teams)

**Pull request merged**